handlers/handlers_test.go

```diff
@@ -29,70 +25,6 @@ func TestOKHandler(t *testing.T) {
 	require.Equal(rr.Body.String(), "OK")
 }
 
-func TestSegmentCallback(t *testing.T) {
```

Comment:
This test needs to be rewritten / split up (along with the underlying code) to test something specific rather than just and E2E call into the HTTP endpoint